pkg/queue/cluster_queue_impl.go

```diff
@@ -83,6 +85,7 @@ func newClusterQueueImpl(
 func (c *clusterQueueBase) Update(apiCQ *kueue.ClusterQueue) error {
 	c.rwm.Lock()
 	defer c.rwm.Unlock()
+	c.queueingStrategy = apiCQ.Spec.QueueingStrategy
```

Comment:
Shouldn't we move all the inadmissible pods back to the heap when changing to StrictFIFO?

Maybe add a unit test? Or integration.

Reply:
> Shouldn't we move all the inadmissible pods back to the heap when changing to StrictFIFO?

This is already happening, because when the `spec.queueingStrategy` is updated, then `specUpdated=true`: https://github.com/kubernetes-sigs/kueue/blob/fa3c1f34662fbbe0ee9411a3d4eed0ae010fdac6/pkg/controller/core/clusterqueue_controller.go#L305-L312

This in turn calls `queueAllInadmissibleWorkloadsInCohort` within cohort here: https://github.com/kubernetes-sigs/kueue/blob/fa3c1f34662fbbe0ee9411a3d4eed0ae010fdac6/pkg/queue/manager.go#L165.

> Maybe add a unit test? Or integration.

This is covered by the integration test. You can see this in this section: https://github.com/kubernetes-sigs/kueue/blob/0ab2501c59d880f7b71376b8c12e3a3663b80bb0/test/integration/scheduler/scheduler_test.go#L1828-L1864. First, we assert `ExpectPendingWorkloadsMetric(0, 1)`, meaning one inadmissible pending. After the change to StrictFIFO we assert `ExpectPendingWorkloadsMetric(1, 0)` meaning it was moved to the heap.

As for the unit test I can try to add but the logic is scattered across `clusterqueue_controller`: `Update` and `Reconcile`, but also manager and cache, so seems involving, and the integration test is needed anyway,

Reply:
Thanks for checking